ambry-store/src/main/java/com.github.ambry.store/IndexSegment.java

```diff
@@ -547,65 +609,12 @@ private void readFromFile(File fileToRead, Journal journal) throws StoreExceptio
-            if (offsetInLogSegment + blobValue.getSize() <= logEndOffset) {
-              index.put(key, blobValue);
-              logger.trace("IndexSegment : {} putting key {} in index offset {} size {}", indexFile.getAbsolutePath(),
-                  key, blobValue.getOffset(), blobValue.getSize());
-              // regenerate the bloom filter for in memory indexes
-              bloomFilter.add(ByteBuffer.wrap(key.toBytes()));
-              // add to the journal
-              if (offsetInLogSegment != -1 && offsetInLogSegment != blobValue.getOriginalMessageOffset()
-                  && blobValue.getOriginalMessageOffset() >= startOffset.getOffset()) {
-                // we add an entry for the original message offset if it is within the same index segment
-                journal.addEntry(new Offset(startOffset.getName(), blobValue.getOriginalMessageOffset()), key);
-              }
-              journal.addEntry(blobValue.getOffset(), key);
-              sizeWritten.addAndGet(key.sizeInBytes() + valueSize);
-              numberOfItems.incrementAndGet();
-              if (offsetInLogSegment + blobValue.getSize() > maxEndOffset) {
-                maxEndOffset = offsetInLogSegment + blobValue.getSize();
-              }
-            } else {
-              logger.info(
-                  "IndexSegment : {} ignoring index entry outside the log end offset that was not synced logEndOffset "
-                      + "{} key {} entryOffset {} entrySize {} entryDeleteState {}", indexFile.getAbsolutePath(),
-                  logEndOffset, key, blobValue.getOffset(), blobValue.getSize(),
-                  blobValue.isFlagSet(IndexValue.Flags.Delete_Index));
-            }
-          }
-          endOffset.set(new Offset(startOffset.getName(), maxEndOffset));
-          logger.trace("IndexSegment : {} setting end offset for index {}", indexFile.getAbsolutePath(), maxEndOffset);
-          long crc = crcStream.getValue();
-          if (crc != stream.readLong()) {
-            // reset structures
-            this.keySize = Key_Size_Invalid_Value;
-            this.valueSize = Value_Size_Invalid_Value;
-            this.endOffset.set(startOffset);
-            index.clear();
-            bloomFilter.clear();
-            throw new StoreException("IndexSegment : " + indexFile.getAbsolutePath() + " crc check does not match",
-                StoreErrorCodes.Index_Creation_Failure);
-          }
-          break;
-        default:
-          throw new StoreException("IndexSegment : " + indexFile.getAbsolutePath() + " invalid version in index file",
-              StoreErrorCodes.Index_Version_Error);
+      version = stream.readShort();
+      if (version == PersistentIndex.VERSION_0 || version == PersistentIndex.VERSION_1) {
+        readFromStream(stream, crcStream, journal);
+      } else {
+        throw new StoreException("IndexSegment : " + indexFile.getAbsolutePath() + " invalid version found ",
```

Comment:
I did give it a thought about it. But code structure was looking bit weird. For both the cases, mostly its same just for the deserialization of lastModifiedTime and resetKey. Anyways, fixing it now. do take a look